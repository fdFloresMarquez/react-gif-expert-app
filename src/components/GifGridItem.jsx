
export const GifGridItem = ({ title, url, isShown}) => {

    if(!isShown){
        return '';
    }
    
    return (
        <div className="card">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}
