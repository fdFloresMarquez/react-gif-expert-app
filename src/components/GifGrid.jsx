import PropTypes from 'prop-types';
import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category, onRemoveGrid }) => {

    const { images, isLoading } = useFetchGifs(category);

    //Oculta los grid y remueve la categoria correspondiente del arreglo.
    const removeGrid = () => {
        onRemoveGrid(category)
    }

    return (
        <>  
            <h3>{category}</h3>

            {
                isLoading
                    ? (<h2>Cargando...</h2>) : null
            }

            <div className="card-grid">
                {
                    images.map((image) => (
                        <GifGridItem
                            key={image.id}
                            {...image}
                        />
                    ))
                }
            </div>
            
            <div>
                <button onClick={removeGrid}>Borrar</button>
            </div>


        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
    onRemoveGrid: PropTypes.func
}
