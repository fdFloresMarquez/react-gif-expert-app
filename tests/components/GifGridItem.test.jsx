import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components/GifGridItem";


describe('Pruebas en GifGridItem', () => {

    const title = "Naruto";
    const url = "https://naruto.com/";

    test('debe de hacer match con el snapshot', () => {

        const { container } = render(<GifGridItem title={title} url={url} isShown={true} />);
        expect(container).toMatchSnapshot();

    });

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {

        render(<GifGridItem title={title} url={url} isShown={true} />);
        // screen.debug();
        // expect( screen.getByRole('img').src ).toBe( url );
        const { src, alt } = screen.getByRole('img')

        expect(alt).toBe(title);
        expect(src).toBe(url);
    });

    test('debe de mostrar el titulo en el componente', () => {

        render(<GifGridItem title={title} url={url} isShown={true} />);
        expect(screen.getByText(title)).toBeTruthy;

    });

})