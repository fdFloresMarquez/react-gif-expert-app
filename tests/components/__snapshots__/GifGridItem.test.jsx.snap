// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Naruto"
      src="https://naruto.com/"
    />
    <p>
      Naruto
    </p>
  </div>
</div>
`;

exports[`Pruebas en GifGridItem debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Naruto"
      src="https://naruto.com/"
    />
    <p>
      Naruto
    </p>
  </div>
</div>
`;
